comp/core/workloadmeta/collectors/util/ecs_util.go

```diff
@@ -0,0 +1,261 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build docker
+
+package util
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/ecs/metadata/v3or4"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// Isv4TaskEnabled returns true if the v4 task metadata collection is enabled
+func Isv4TaskEnabled() bool {
+	return config.Datadog.GetBool("ecs_metadata_use_v4") && (flavor.GetFlavor() == flavor.DefaultAgent)
```

Comment:
the feature is linked to v4 or it requires v4?
Can we automatically detect if v4 is available? and default to v4 by default? 
I'm asking to know if we should expose in the configuration parameter the `v4` information or just `ecs_task_collection_enabled`

Reply:
I think `ecs_task_collection_enabled` makes more sense. We use v4 as it's the latest endpoint

I added more comments to explain this flag
https://github.com/DataDog/datadog-agent/blob/66b205de2f9722a8c203d30fae8c5a1d10514779/comp/core/workloadmeta/collectors/util/ecs_util.go#L26-L32

I've also added a way to detect version
https://github.com/DataDog/datadog-agent/blob/66b205de2f9722a8c203d30fae8c5a1d10514779/comp/core/workloadmeta/collectors/internal/ecs/ecs.go#L110-L128